src/lib/agent.ts

```diff
@@ -62,13 +69,71 @@ export class Agent {
     eventbus.on("didGenerated", callback)
   }
 
+  private handleDiscoverFeatures(message: IMessage) {
+    const regexEscape = (s: string) => s.replace(/([.*+?$^=!:{}()|\[\]\/\\])/g, "\\$1");
+    const createRegex = (query: string) => (new RegExp(`^${query.split("*").map(regexEscape).join(".*")}$`));
+    let protocolResponse: object[] = [];
+
+    // Loop through all queries, then all implemented protocols and build up a
+    // list of supported protocols that match the user's request
+    for(let query of message.body.queries) {
+
+      // Rudimentary implementation, ignoring all except protocol requests
+      if(query["feature-type"] != "protocol")
+        continue
+
+      for(let protocol of IMPLEMENTED_MESSAGE_TYPES) {
+        if(createRegex(query["match"]).test(protocol)) {
+          protocolResponse.push({
+            "feature-type": "protocol",
+            "id": protocol,
+          })
+        }
+      }
+    }
+    const response = {
+      "type": "https://didcomm.org/discover-features/2.0/disclose",
+      "thid": message.id,
+      "body": {
+        "disclosures": protocolResponse
+      }
+    }
+    return response
+  }
+
+  private handleCoreProtocolMessage(message: IMessage) {
+    switch(message.type) {
+      case "https://didcomm.org/trust-ping/2.0/ping":
+        if(message.body?.response_requested !== false) {
+          this.sendMessage({
+              did: message.from
+            }, {
+            "type": "https://didcomm.org/trust-ping/2.0/ping-response",
+            "thid": message.id,
+          })
```

Comment:
I think we should modify the `sendMessage` method to accept a union of `Contact | DID` (where DID is a hypothetical alias to just string) so we can call this method like:

```ts
this.sendMessage(message.from, { ... })
```

Just to make it a bit more readable and developer friendly.